core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/StackOperationService.java

```diff
@@ -264,21 +282,19 @@ public FlowIdentifier updateNodeCountStopStart(Stack stack, InstanceGroupAdjustm
                             instanceGroupAdjustmentJson,
                             stackWithLists,
                             instanceGroupAdjustmentJson.getScalingAdjustment());
+                    // TODO CB-14929: This validations needs to be removed or adjusted. Checks for UNHEALTHY service status. Needs to change to check unhealthy CM / unhealthy RM eventually.
                     updateNodeCountValidator.validataHostMetadataStatuses(stackWithLists, instanceGroupAdjustmentJson);
                 }
-                if (instanceGroupAdjustmentJson.getScalingAdjustment() > 0) {
-                    // TODO CB-14929: When is the DB update performed? Should this be before or after a flow is accepted. The next step sends a trigger.
-                    stackUpdater.updateStackStatus(stackWithLists.getId(), DetailedStackStatus.UPSCALE_REQUESTED,
-                            "Requested node count for upscaling (stopstart): " + instanceGroupAdjustmentJson.getScalingAdjustment());
-                    return flowManager.triggerStopStartStackUpscale(
-                            stackWithLists.getId(),
-                            instanceGroupAdjustmentJson,
-                            withClusterEvent);
-                } else {
-                    stackUpdater.updateStackStatus(stackWithLists.getId(), DetailedStackStatus.DOWNSCALE_REQUESTED,
-                            "Requested node count for downscaling: " + abs(instanceGroupAdjustmentJson.getScalingAdjustment()));
-                    return flowManager.triggerStackDownscale(stackWithLists.getId(), instanceGroupAdjustmentJson);
-                }
+                // TODO CB-14929: Post CB-15162 and add an additional check to ignore upscale requests of size 0 in stackCommonService
+
+                // TODO CB-14929: When should DB status be updated. Should the DetailStackStatus be something other than UPSCALE_REQUESTED. i.e. differentiate between
+                //  an actual upscale and a 'VM pool / start-stop'
+                stackUpdater.updateStackStatus(stackWithLists.getId(), DetailedStackStatus.UPSCALE_REQUESTED,
```

Comment:
note to myself : look for all the places where node, cluster and stack status is updated for scale up / down, and start up / down.